packages/url_launcher/url_launcher_web/lib/link.dart

```diff
@@ -0,0 +1,295 @@
+
+/// The unique identifier for the view type to be used for link platform views.
+const String viewType = '__url_launcher::link';
+
+/// The name of the property used to set the viewId on the DOM element.
+const String _viewIdProperty = '__url_launcher::link::viewId';
+
+// TODO(mdebbar): Should use PlatformViewFactory.
+dynamic get webViewFactory => _LinkViewController._viewFactory;
+
+final MethodCodec _codec = const JSONMethodCodec();
+
+/// The delegate for building the [Link] widget on the web.
+///
+/// It uses a platform view to render an anchor element in the DOM.
+class WebLinkDelegate extends StatefulWidget {
+  /// Creates a delegate for the given [link].
+  const WebLinkDelegate(this.link);
+
+  /// The [Link] widget built by the app.
+  final Link link;
+
+  @override
+  _WebLinkDelegateState createState() => _WebLinkDelegateState();
+}
+
+class _WebLinkDelegateState extends State<WebLinkDelegate> {
+  _LinkViewController _controller;
+
+  @override
+  void didUpdateWidget(WebLinkDelegate oldWidget) {
+    super.didUpdateWidget(oldWidget);
+    if (widget.link.uri != oldWidget.link.uri) {
+      _controller?._setUri(widget.link.uri);
+    }
+    if (widget.link.target != oldWidget.link.target) {
+      _controller?._setTarget(widget.link.target);
+    }
+  }
+
+  Future<void> _followLink() {
+    final Completer<void> completer = Completer<void>();
+    _LinkViewController._registerHitTest(
+      _controller,
+      onClick: (html.MouseEvent event) {
+        completer.complete(_onDomClick(event));
+      },
+    );
+    return completer.future;
+  }
+
+  Future<void> _onDomClick(html.MouseEvent event) {
+    if (!widget.link.uri.hasScheme) {
+      // A uri that doesn't have a scheme is an internal route name. In this
+      // case, we push it via Flutter's navigation system instead of letting the
+      // browser handle it.
+      event.preventDefault();
+      final String routeName = widget.link.uri.toString();
+      // TODO(mdebbar): how do we know if this is true or false?
+      final bool isUsingRouter = false;
+      // QUESTION: does this update the history entry on the engine side?
+      final MethodCall call = isUsingRouter
+          ? MethodCall('pushRouteInformation', <dynamic, dynamic>{
+              'location': routeName,
+              'state': null,
+            })
+          : MethodCall('pushRoute', routeName);
+      return pluginBinaryMessenger.send(
+        'flutter/navigation',
+        _codec.encodeMethodCall(call),
+      );
+    }
+
+    // External links will be handled by the browser, so we don't have to do
+    // anything.
+    return Future<void>.value(null);
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Stack(
+      children: <Widget>[
+        widget.link.builder(
+          context,
+          widget.link.isDisabled ? null : _followLink,
+        ),
+        Positioned.fill(
+          // child: _LinkHtmlView(uri: link.uri, target: link.target),
```

Comment:
delete this line and add a TODO to use a `PlatformView` once hit testing is fixed and creation parameters supported